### ISSUE TITLE
Travis failures.

### ISSUE THREAD

**Issue by elmart** [neovim/neovim#2519]
It does nothing and fails because no output again and again for me.

Also, just for curiosity, when have we removed @jszakmeister's QuickBuild? I hadn't realized.

**Comment by jszakmeister** [comment]
The QB issue is probably mine.  I had to change my token, so there may be a spot that I forgot.  I'll look into it.

**Comment by elmart** [comment]
QuickBuild is working again.

BTW, I take the chance to congratulate you because of it. It has a bit old-fashioned interface, but it does a really good work, and it does it really quick. I'm curious how long it would take if we included everything Travis does now (maintaining the extra things it does now (freebsd, release)). Have you tried that, or is it there any roadblock? Travis is getting really slow and failure-prone lately...

**Comment by fwalch** [comment]
@elmart FWIW, some of the Travis problems might be because they're phasing out support for non-containerized builds (ref #2340).

**Comment by elmart** [comment]
@fwalch Yes, it's definitely something about their infrastructure that is gone rotten.
Now it has failed again for me. Build completes (I see the "ALL DONE." line meaning old tests have finished successfully), but whatever it tries to do after that, fails with:
```
Invalid .gcno File!
Unexpected version: *904.
Invalid .gcno File!
Unexpected version: *904.
Invalid .gcno File!
Unexpected version: *904.
Invalid .gcno File!
Unexpected version: *904.
Invalid .gcno File!
/Library/Python/2.7/site-packages/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
{'error': u'Failure to submit data. Response [504]: <!DOCTYPE html>\n<html lang="en">\n\n  <head>\n<meta http-equiv="refresh" content="0">\n\n<meta http-equiv="set-cookie" content="cf_use_ob=443; expires=Mon, 27-Apr-15 18:21:07 GMT; path=/">\n\n<meta http-equiv="set-cookie" content="cf_ob_info=504:1ddc8a23d7c60293:SJC; expires=Mon, 27-Apr-15 18:21:07 GMT; path=/">\n\n\n    <meta charset="utf-8" />\n    <meta content="IE=Edge,chrome=1" http-equiv="X-UA-Compatible" />\n    <title>Coveralls :: Oops, something went wrong (500)</title>\n\n    <style type="text/css">@font-face{font-family:\'Open Sans\';font-style:normal;font-weight:400;src:local(\'Open Sans\'),local(\'OpenSans\'),url(data:font/woff;base64,d09GRgABAAAAADkMAA8AAAAAYjgAAQABAAAAAAAAAAAAAAAAAAAAAAAAAABHREVGAAABWAAAABwAAAAcAOYBo0dQT1MAAAF0AAAADAAAAAwAFQAKR1NVQgAAAYAAAAAMAAAADAAVAApPUy8yAAABjAAAAF8AAABgoT6eyWNtYXAAAAHsAAAAkAAAALyNxZDlZ2FzcAAAAnwAAAAQAAAAEAAVACNnbHlmAAACjAAAJC4AADQIMQ5sJGhlYWQAACa8AAAANgAAADb8lOQCaGhlYQAAJvQAAAAfAAAAJA3MBqNobXR4AAAnFAAAAgcAAANUkLZ      IYX2tlcm4AACkcAAAKEgAAHpCj66TjbG9jYQAAMzAAAAGwAAABsAuUF0BtYXhwAAA04AAAACAAAAAgAnACCm5hbWUAADUAAAACnwAABcdzsIiFcG9zdAAAN6AAAAFsAAAB+cBBVBwAAQAAAAwAAAAAAAAAAgACAAAAzwABANEA1gABAAEAAAAKAAoACgAAAAEAAAAKAAoACgAAeAFjYGbZxjiBgZWBg3UWqzEDA6M8hGa+yJDG+JGDiYmbjY2ZlYWJieUBA9N7BwaFaAYGBg0GIDB0DHZmAAr8/8sm/0+EoYWjlylCgYFxPkiOxYN1G5BSYGAGAPjQEDkAeAFjYGBgZGAGYgYGHgYWhg1AWoNBAcjiYPBg8GSoYzjNcJ7hP6MhYzDTMaZbTHcURBSkFOQUlBSsFFwUShTW/P8P1KEAVOnFsIDhDMMFoMogqEphBQkFGaBKS5jK/48ZOv8/+n+Qgf3/3v89/3P+/vr7/O+zBxserH2w5sHKB5MfdD2IfaAMdAuRAAAFlTOhAAEAAwAIAAoADQAH//8AD3gBhboLXIzp+z/+XPfzzAGxHaaG7YPGmEYINU1TSKSSnJK0pE2SlFNSSdq0apOElkhFjklGi22Tw8YW6xBLa53X2r7YZe36tFi7rGbu/vf9PDOaz378f596Pc10Pc993dfpft/XfV0Pg5hGhuE6i+oYlpEwjJ3CRqEjVyM71nA0HqUb80R1b4LjuUcM+UFMSftDyOef7cYwoGQ1rIfG3cFeJlb2cYbABE3z/WVDfUd6e/hDEad8U7s60G/kGF8yLpetRIfNcwCrZBXkAt9+CS5ogEu8i6jO+AxZ04s8qyHy/EWedWR602cV9uRZO/7SKvhLw/KXPSgI+YYfOOLLYflh+Gro6hD8Bnr7499gQGhBKLiFrQoDqeFnGOyHr7I5uHolDoWD9FoJYVlQi8fTKwtXQxhDJMtpz+esxLaME+PMuJKZZWKJvVLbx1mtdXDQ2Dg7az08dVqNvYNc4qy26YUkHuTDE2QOchuxmLP65ujKxd+NnnJn+uW9l/atPHbAY3NZ+bZg/fSP7xjvRSTOjoPG1Z/Lf2lR9t6mGgzHRlWvzqmyrasVBeQM64Inus/MmBM0fSDO6MVKxke6QI71h9TeIia+/anYVXSJ6cTYMwoilRvxkMxB425jrewjFlGzU7GIC4BSPbUezv+kgxJeBE+pqJgSDBdLN63dunnjhm1QGRwWFhISFhYMl0o3rS/dvHH9DowNN4rYARzS6yEMQqv0Pz1pbXn4qNVw98D+fZ8d2Lv3wMMnrT8+fPQb6/QmmPdkUvtT0TXRZaYLkU3LMNQSniYJ7Nx10A3ZyxxUxGaDkCCKPTEoeHrqNEgCSrWYD5ntQR+snDM3fHri1uc78LjEyIFb8fGC2qk+fb8+tPvE6nLY4jVaXuWfDwN+Pr70ZfGtf3NFAZnTxmVNmTAruq28DKr8p8eNSs1/k30hbmbMPO/iqn2b5x/5EKeP2DcbP9iE79XMi7xO7Ag0/sCXl5oBc+QJQcfw90NxPVKQ+12Jfa1tdRoxEd5WrnRGoVs37v50w6aCnUVlyA06wZWDjdj95TPseVIP54SxPmSslTAWrJFE6Wmr9UBqjYMtstq6cWfBpg2f7qaD8d94aFU9XHr2Eq40HsJu/NhwlMV1E8v41cSq5CI7Ccuq7VQ6EYs2DID17+Oc19WHdx5+gfN6Qt4AsQynJtb1xieiYB4ujoLA3nWJUED5xDMPORfuDPEFA2TB2ChtyEKx0aB1sA3HFuA4KCtg5WtwOOjXQDV53he/hoVMKyNlGJVczAutg6j33o/RZjqO6toqj8WvlsSDeyTlHQZ3kC9Koraz0yrsw+A3uFNSIujO4wHzgs6rMzuZ+LXEx3voKD9vzeh5owMCRo8K9KXPyhgG3TX5AJSA7hpbKqnRaThRbIlrf8q58vEk5y2p7INsrG017rbA/7XhKZzr7389/av1Zesrw/3iisrNmysritE9nIvXQBYsgRWwBK/AG3EjvgdqGEZ+VbiFzF1PJrhEpuncoveralls upload fail/Users/travis/build/neovim/neovim/.ci/gcc.sh: line 42: echo: write error: Resource temporarily unavailable
ed.
The command "sudo -E su $USER -c "sh -e \"$CI_SCRIPTS/$CI_TARGET.sh\""" exited with 1.
Done. Your build exited with 1.
```
No idea.

**Comment by grtlr** [comment]
According to their ([status page](http://www.traviscistatus.com/)) they are pretty much at their limit for current non-containerized open source builds. Maybe that is why the builds take longer at the moment. In the container-based setup on the other hand, there still seems to be some capacity.

**Comment by elmart** [comment]
Now it failed build for last commit I merged into master: https://travis-ci.org/neovim/neovim/builds/60342107.
Failure is because one of those non-deterministic tests. PR passed ok. But when I try to restart failed job, it doesn't really do it.

**Comment by jszakmeister** [comment]
> BTW, I take the chance to congratulate you because of it. It has a bit old-fashioned interface, but it does a really good work, and it does it really quick. I'm curious how long it would take if we included everything Travis does now (maintaining the extra things it does now (freebsd, release)). Have you tried that, or is it there any roadblock? Travis is getting really slow and failure-prone lately...

Sorry for the slow response... some things have gone into production this week, and I've been helping to iron out the wrinkles.

Thanks @elmart!  It was part of my intention to have things run quicker than the current Travis builds, and provide as much feedback as we can as quickly as we can so that PR authors don't have to wait 40 minutes to get feedback.  Honestly, I have not tried running full up everything (code coverage, etc.).  Most of it I think would be fine, but there are a couple of roadblocks.  We could certainly add running of the legacy test suite, add code coverage, and those things.  One headache I see is that QB doesn't have a great way of working with credentials for external services like Coveralls.  There are some things we can do, but none of them are great and generally mean any administrator of the instance would have have direct access to a token. :-(

I (we?) also need to think about reporting on some bits.  I'd rather not attempt to collect and report code coverage on every build machine, but just limit it to one platform for now.  The problem is that it could mean we miss something important, say in Windows, if/when we get a Windows VM running under QB.  This also potentially puts a bit more strain on a VM, but I think it'd be okay.  The one thing that is nice about this setup is that the infrastructure is not being shared with anything else.  It's wholly dedicated to Neovim.

Also, right now, I'm the only one in control of the build steps for QB, which makes it a bit of a problem if you guys feel something needs to be done differently.  I need to get a new Mac OS X VM stood up, and then I'd be willing to offer others some access to it.  We could push some of that into the repo's .ci area, but we wouldn't want to push all of it there.  We need to be careful though--a mistake here could mean having to rebuild the VM.  Being able to put separate steps in QB means that we can adjust timeouts for particular steps, allow steps to run in parallel on a bunch of different machines, etc.

I also have a no longer used machine from work that I'd like to get some VMs installed onto it as backups for the current VMs--that way if my box goes offline, there are still other VMs that can do the builds.  The machine is in good shape, I just need some time to get things going and that's been very hard to come by lately. :-)

**Comment by glts** [comment]
The Linux `gcc` build now times out more often than not (> 50min). In consequence, the majority of pull requests now get the red X for build failure.

It has been like this for a few days now. What can we do about this?

**Comment by fwalch** [comment]
See #2678.

**Comment by justinmk** [comment]
Should be fixed now.
